In [1]:
import pandas as pd
year0 = pd.read_csv("Data/final_tmdb_data_2000.csv.gz", low_memory = False)
year1 = pd.read_csv("Data/final_tmdb_data_2001.csv.gz", low_memory = False)

In [2]:
data = [year0, year1]
combined_data = pd.concat(data)
combined_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2631 entries, 0 to 1423
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                2631 non-null   object 
 1   adult                  2629 non-null   float64
 2   backdrop_path          1378 non-null   object 
 3   belongs_to_collection  202 non-null    object 
 4   budget                 2629 non-null   float64
 5   genres                 2629 non-null   object 
 6   homepage               194 non-null    object 
 7   id                     2629 non-null   float64
 8   original_language      2629 non-null   object 
 9   original_title         2629 non-null   object 
 10  overview               2578 non-null   object 
 11  popularity             2629 non-null   float64
 12  poster_path            2328 non-null   object 
 13  production_companies   2629 non-null   object 
 14  production_countries   2629 non-null   object 
 15  rele

# 1. Valid Financial Information

In [3]:
combined_data2 = combined_data[(combined_data.budget > 0) | 
                                (combined_data.revenue > 0)]
combined_data2.shape

(676, 26)

### 676 movies had some financial information

# 2. How many movies in each certification categories

In [4]:
combined_data2["certification"].value_counts()

R        250
PG-13    147
PG        41
NR        17
G         14
Name: certification, dtype: int64

### 14 movies with G rating. 41 movies with PG rating. 147 movies with PG-13 rating. 250 movies with R rating 

# 3. Average revenue per certification category

In [5]:
g_movies = combined_data2.loc[(combined_data2["certification"] == "G")]
print(f"Average revenue for G rated movies: {g_movies.revenue.mean():.2f}")
pg_movies = combined_data2.loc[(combined_data2["certification"] == "PG")]
print(f"Average revenue for PG rated movies: {pg_movies.revenue.mean():.2f}")
pg13_movies = combined_data2.loc[(combined_data2["certification"] == "PG-13")]
print(f"Average revenue for PG-13 rated movies: {pg13_movies.revenue.mean():.2f}")
r_movies = combined_data2.loc[(combined_data2["certification"] == "R")]
print(f"Average revenue for R rated movies: {r_movies.revenue.mean():.2f}")

Average revenue for G rated movies: 123708547.43
Average revenue for PG rated movies: 107284486.54
Average revenue for PG-13 rated movies: 93643076.01
Average revenue for R rated movies: 33236616.42


# 4. Average budget per certification category

In [6]:
print(f"Average budget for G rated movies: {g_movies.budget.mean():.2f}")
print(f"Average budget for PG rated movies: {pg_movies.budget.mean():.2f}")
print(f"Average budget for PG-13 rated movies: {pg13_movies.budget.mean():.2f}")
print(f"Average budget for R rated movies: {r_movies.budget.mean():.2f}")

Average budget for G rated movies: 40857142.86
Average budget for PG rated movies: 44229268.51
Average budget for PG-13 rated movies: 41482366.85
Average budget for R rated movies: 20678017.80
